.githooks/githooks.test.sh

```
@@ -97,24 +97,25 @@ write_ref_updates_file() {  # repo remote local_ref remote_ref output_file
   printf '%s %s %s %s\n' "$local_ref" "$local_sha" "$remote_ref" "$remote_sha" > "$output_file"
 }
 
+# Extracted from the real pre-push, not hand-duplicated: a second copy of this
+# selection algorithm is exactly how it went stale before (see new_ref_base_sha's
+# own history) — a fix applied to only one copy leaves the other asserting
+# against a base the real hook no longer picks.
+eval "$(sed -n '/^new_ref_base_sha() {/,/^}/p' "$REPO_ROOT/.githooks/pre-push" | \
+  sed -e 's|\${repo_root}|${repo}|g' -e 's|\$repo_root|\$repo|g' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)